### PR TITLE
(TopologySpread) Evict pods with selectors that match multiple nodes

### PR DESCRIPTION
**Pull request by damemi**
Fixes https://github.com/kubernetes-sigs/descheduler/issues/524

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
/cc @seanmalloy @ingvagabund

**Review by uthark — APPROVED** [review]

File: pkg/descheduler/strategies/topologyspreadconstraint.go [hunk 1]
```diff
@@ -289,11 +289,11 @@ func balanceDomains(
 			// however we still account for it "being evicted" so the algorithm can complete
 			// TODO(@damemi): Since we don't order pods wrt their affinities, we should refactor this to skip the current pod
 			// but still try to get the required # of movePods (instead of just chopping that value off the slice above)
-			if aboveToEvict[k].Spec.NodeSelector != nil ||
```
Comment (ingvagabund) [inline pkg/descheduler/strategies/topologyspreadconstraint.go:292]: Can you break the changes into two steps?
1. create `isRequiredDuringSchedulingIgnoredDuringExecution = aboveToEvict[k].Spec.Affinity != nil && aboveToEvict[k].Spec.Affinity.NodeAffinity != nil && aboveToEvict[k].Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil` variable so it's easier to read the condition
2. fix the bug

In which case the change will look like:

```go
aboveToEvict[k].Spec.NodeSelector != nil ||
(
	isRequiredDuringSchedulingIgnoredDuringExecution &&
	nodesPodFitsOnBesidesCurrent(aboveToEvict[k], nodeMap) == 0
)
```
-->
```go
(
	aboveToEvict[k].Spec.NodeSelector != nil || isRequiredDuringSchedulingIgnoredDuringExecution
) &&
nodesPodFitsOnBesidesCurrent(aboveToEvict[k], nodeMap) == 0
```

After which it's clear why the change is required.

**Review by ingvagabund — CHANGES_REQUESTED** [review]
Small change to improve the readability

**Comment by damemi** [comment]
Thanks @ingvagabund, updated

**Pull request merged**